catalog/imaging/imaging-postprocess-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostProcessPlugin.java

```diff
@@ -216,49 +233,86 @@ private void handleProcessCreateItem(List<ProcessCreateItem> processCreateItems)
   }
 
   private Stream<ProcessCreateItem> handleProcessCreateItem(ProcessCreateItem processCreateItem) {
-    List<ProcessCreateItem> createdItems = new ArrayList<>();
+    List<ProcessCreateItem> createdItems = null;
     Metacard metacard = processCreateItem.getMetacard();
     ProcessResource processResource = processCreateItem.getProcessResource();
 
     if (shouldProcess(processResource)) {
-      try (TemporaryFileBackedOutputStream fbos =
-          new TemporaryFileBackedOutputStream(DEFAULT_FILE_BACKED_OUTPUT_STREAM_THRESHOLD)) {
-
-        // TODO: 06/21/2018 oconnormi - can probably get rid of this once
-        // ProcessResourceImpl.getInputStream can be called multiple times
-        fbos.write(IOUtils.toByteArray(processResource.getInputStream()));
-        ByteSource byteSource = fbos.asByteSource();
-        BufferedImage renderedImage = renderImage(byteSource.openStream());
-
-        if (renderedImage != null) {
-          addThumbnailToMetacard(metacard, renderedImage);
-          processCreateItem.markMetacardAsModified();
-          if (createOverview) {
-            ProcessResource overviewProcessResource =
-                createOverviewResource(renderedImage, metacard);
-            createdItems.add(new ProcessCreateItemImpl(overviewProcessResource, metacard));
-          }
-
-          if (storeOriginalImage) {
-            ProcessResource originalImageProcessResource =
-                createOriginalImage(
-                    renderImageUsingOriginalDataModel(byteSource.openStream()), metacard);
 
-            createdItems.add(new ProcessCreateItemImpl(originalImageProcessResource, metacard));
-          }
+      try {
+        lock.acquire();
+        try {
+          createdItems =
+              process(
+                  processCreateItem,
+                  metacard,
+                  null,
+                  processResource,
+                  constructorTriple ->
```

Comment:
❓ Is the code uglier if you change this to a Pair instead of a triple and overload the `process` method to take either a pair or a triple?